src/charm.py

```diff
@@ -188,6 +178,10 @@ def __init__(self, *args):
         )
         # Handling config changed event to publish the new url if the unit reboots and gets new IP
         self.framework.observe(self.on.config_changed, self._configure_sdcore_nms)
+        self.framework.observe(
+            self.on["nms"].pebble_custom_notice, self._sync_network_config
```

Comment:
POSTing config directly (through cURL, Swagger, Postman, ...) will trigger that. If there's no relations in the model, the code will do nothing, because the first step of syncing network config is going over the relations. Once the relation is properly created, the main callback will be called and it also runs `self._sync_network_config` to cover "funny" cases like this one.